docs/examples/use_cases/tensorflow/efficientdet/train.py

```diff
@@ -0,0 +1,232 @@
+            image_size = params["image_size"]
+            model.predict(np.zeros((1, image_size[0], image_size[1], 3)))
+            model.load_weights(args.start_weights)
+            fname = args.start_weights.split("/")[-1]
+            ckpt_pattern = f"{args.model_name}\.(\d\d+)\.h5"
+            match = re.match(ckpt_pattern, fname)
+            if match:
+                initial_epoch = int(match.group(1).lstrip("0"))
+
+        callbacks = []
+
+        if args.ckpt_dir:
+            ckpt_dir = args.ckpt_dir
+            if not tf.io.gfile.exists(ckpt_dir):
+                tf.io.gfile.makedirs(tensorboard_dir)
+            callbacks.append(
+                tf.keras.callbacks.ModelCheckpoint(
+                    filepath=os.path.join(
+                        ckpt_dir, "".join([args.model_name, ".{epoch:02d}.h5"])
+                    ),
+                    save_weights_only=True,
+                )
+            )
+
+        if args.log_dir:
+            log_dir = args.log_dir
+            if not tf.io.gfile.exists(log_dir):
+                tf.io.gfile.makedirs(log_dir)
+            callbacks.append(
+                tf.keras.callbacks.TensorBoard(log_dir=log_dir, update_freq="epoch")
+            )
+
+        model.fit(
+            train_dataset,
+            epochs=args.epochs,
+            steps_per_epoch=args.train_steps,
+            initial_epoch=initial_epoch,
+            callbacks=callbacks,
+            validation_data=eval_dataset if args.eval_during_training else None,
+            validation_steps=args.eval_steps,
+            validation_freq=args.eval_freq,
+        )
+
+        if args.eval_after_training:
+            print("Evaluation after training:")
+            model.evaluate(eval_dataset, steps=args.eval_steps)
+
+        model.save_weights(args.output)
+
+
+if __name__ == "__main__":
+    import argparse
+    from argparse_utils import enum_action
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--initial_epoch",
+        type=int,
+        default=0,
+        help="epoch from which to start training",
+    )
+    parser.add_argument(
+        "--epochs", type=int, default=300, help="epoch on which training should finish"
+    )
+    parser.add_argument(
+        "--input",
+        action=enum_action(utils.InputType),
+        required=True,
+        help="input type",
+    )
+    parser.add_argument(
+        "--images_path",
+        required=False,
+        help="Path to COCO images",
+    )
+    parser.add_argument(
+        "--annotations_path",
+        required=False,
+        help="Path to COCO annotations",
+    )
+    parser.add_argument(
+        "--train_file_pattern",
+        required=False,
+        help="glob pattern for TFrecord files with training data",
+    )
+    parser.add_argument("--train_batch_size", type=int, default=64)
+    parser.add_argument(
+        "--train_steps", type=int, default=2000, help="number of steps in each epoch"
```

Comment:
What are the `steps` here? If these are meant to be `iterations`, shouldn't it be inferred from the batch size and dataset size?

Reply:
Steps are indeed iterations, I'm using Keras naming convention here.
In order to infer iterations number from batch size and dataset size, we'd have to ask the user for dataset size (num_examples_per_epoch argument in original EfficientDet implementation) and I think that asking here for the number of steps is better as it is easier to run training with a small number of steps for example for testing purposes.